### ISSUE TITLE
Embedding Layers error in Similarity `_TensorFlowBackend` 

### ISSUE THREAD

**Issue by mauicv** [SeldonIO/alibi#828]
The following raises the error `'IndexedSlices' object has no attribute 'numpy'`:

```py
model = tf.keras.models.Sequential([
        tf.keras.layers.Embedding(10, 4, input_shape = (5,)),
        tf.keras.layers.Flatten(), 
        tf.keras.layers.Dense(1)
    ])
    X = tf.random.uniform(shape = (1, 5), minval = 0, maxval = 10, dtype = tf.float32)
    Y = tf.random.uniform(shape = (1, 1), minval = 0, maxval = 10, dtype = tf.float32)
    loss_fn = tf.keras.losses.MeanSquaredError()
    tf_grads = _TensorFlowBackend.get_grads(model, X, Y, loss_fn)
```

<s>This is because the gradient through an embedding layer is not computable.</s> The gradients get returned as `IndexedSlices` which are sparse representations. These raise an error in `explainers.similarity.backends.tensorflow.base` in the following code:

```py
        with tf.device(_TensorFlowBackend.device):
            with tf.GradientTape() as tape:
                output = model(X, training=False)
                loss = loss_fn(Y, output)

            # compute gradients of the loss w.r.t the weights
            grad_X_train = tape.gradient(loss, model.trainable_weights)
            grad_X_train = np.concatenate([w.numpy().reshape(-1) for w in grad_X_train])
```

because if `w` is an instance of `IndexedSlices` it has no `numpy` method.
